bears/go/GoVetBear.py

```diff
@@ -25,5 +25,130 @@ class GoVetBear:
+                         struct_tags: bool=False,
+                         tests_and_docs: bool=False,
+                         unreachable: bool=False,
+                         unsafe_pointers: bool=False,
+                         unused_result: bool=False):
+        """
+        :param assembly_declarations:
+            Mismatches between assembly files and Go function declarations.
+        :param assign:
+            Check for useless assignments.
+        :param atomic:
+            Common mistaken usages of the sync/atomic package.
+        :param boolean:
+            Common mistaken usages of the sync/atomic package.
+        :param build_tags:
+            Mistakes involving boolean operators.
+        :param invalid_cgo:
+            Detect some violations of the cgo pointer passing rules.
+        :param unkeyed_composite_literals:
+            Composite struct literals that do not use the field-keyed syntax.
+        :param copying_locks:
+            Locks that are erroneously passed by value.
+        :param http_response:
+            Mistakes deferring a function call on an HTTP response before
+            checking whether the error returned with the response was nil.
+        :param lost_cancel:
+            The cancelation function returned by context.WithCancel,
+            WithTimeout, and WithDeadline must be called or the new
+            context will remain live until its parent context is cancelled.
+        :param methods:
+            Non-standard signatures for methods with familiar names, including:
+            Format, GobEncode, GobDecode, MarshalJSON, MarshalXML, Peek,
+            ReadByte,ReadFrom, ReadRune, Scan, Seek, UnmarshalJSON, UnreadByte,
+            UnreadRune, WriteByte, WriteTo
+        :param nil_function:
+            Comparisons between functions and nil.
+        :param printf:
+            Suspicious calls to functions in the Printf family.
+        :param range_loops:
+            Incorrect uses of range loop variables in closures.
+        :param shadow_variables:
+            Variables that may have been unintentionally shadowed.
+        :param shifts:
+            Shifts equal to or longer than the variable's length.
+        :param struct_tags:
+            Struct tags that do not follow the format understood by
+            reflect.StructTag.Get. Well-known encoding struct tags (json, xml)
+            used with unexported fields.
+        :param tests_and_docs:
+            Mistakes involving tests including functions with incorrect names
+            or signatures and example tests that document identifiers not in
+            the package.
+        :param unreachable:
+            Unreachable code.
+        :param unsafe_pointers:
+            Likely incorrect uses of unsafe.Pointer to convert integers to
+            pointers.
+        :param unused_result:
+            Calls to well-known functions and methods that return a value that
+            is discarded
```

Comment:
The setting names should be a bit clearer, like when you enable the setting, what constructs are allowed / forbidden.

Reply:
I've used the same names that are a part of Vet itself. So, someone familiar with Vet will be able to map his understanding with the functionalities of this bear. Using different names IMO can prove to be conflicting for users with the working knowledge of only Vet or both Vet and GoVetBear. Thoughts?

Reply:
It should resemble coala's style, as those are then part of coala's native settings set. If the names are well-phrased, they can be used across different bears for the same things.

Reply:
I've changed the names of almost all of the params. Still, if they seem to be less self-explanatory then that may be because most of them are Go specific. If you go through their explanations, they look justified. Let me know if anything else needs to be changed. ;)